src/main/java/com/onwelo/tinvalidator/validator/IEValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.onwelo.tinvalidator.validator;
+
+import java.util.regex.Pattern;
+
+public class IEValidator implements Validator {
+
+    private static final Pattern PATTERN = Pattern.compile("\\d{7}[A-W][A-IW]?");
```

Comment:
According to new document pattern should be different. There are two style 2 cover. First has 7 digits and a number on the end, second has 8 signs where 2nd and 8th are characters and the rest are numbers.